diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d38212c..8c294429 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,7 @@ endif()
 
 # === Build path ===
 
-set(LLGL_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/build" CACHE PATH "Override binary output directory")
+# set(LLGL_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/build" CACHE PATH "Override binary output directory")
 
 set( EXTERNAL_MODULE_DIR            "${PROJECT_SOURCE_DIR}/cmake"                   )
 set( PROJECT_INCLUDE_DIR            "${PROJECT_SOURCE_DIR}/include"                 )
diff --git a/include/LLGL/Backend/Metal/NativeHandle.h b/include/LLGL/Backend/Metal/NativeHandle.h
index de87278d..ae5f02c9 100644
--- a/include/LLGL/Backend/Metal/NativeHandle.h
+++ b/include/LLGL/Backend/Metal/NativeHandle.h
@@ -8,8 +8,9 @@
 #ifndef LLGL_METAL_NATIVE_HANDLE_H
 #define LLGL_METAL_NATIVE_HANDLE_H
 
-
+#ifdef __OBJC__
 #import <Metal/Metal.h>
+#endif
 
 
 namespace LLGL
@@ -26,7 +27,11 @@ namespace Metal
 */
 struct RenderSystemNativeHandle
 {
+#ifdef __OBJC__
     id<MTLDevice> device;
+#else
+    void* device;
+#endif
 };
 
 /**
@@ -35,7 +40,11 @@ struct RenderSystemNativeHandle
 */
 struct CommandBufferNativeHandle
 {
+#ifdef __OBJC__
     id<MTLCommandBuffer> commandBuffer;
+#else
+    void* commandBuffer;
+#endif
 };
 
 /**
@@ -77,6 +86,7 @@ struct ResourceNativeHandle
     
     union
     {
+#ifdef __OBJC__
         //! Specifies the native Metal MTLBuffer object.
         id<MTLBuffer>       buffer;
 
@@ -85,6 +95,16 @@ struct ResourceNativeHandle
 
         //! Specifies the native Metal MTLSamplerState object.
         id<MTLSamplerState> samplerState;
+#else
+        //! Specifies the native Metal MTLBuffer object.
+        void* buffer;
+
+        //! Specifies the native Metal MTLTexture object.
+        void* texture;
+
+        //! Specifies the native Metal MTLSamplerState object.
+        void* samplerState;
+#endif
     };
 };
 
diff --git a/include/LLGL/Backend/OpenGL/MacOS/MacOSNativeHandle.h b/include/LLGL/Backend/OpenGL/MacOS/MacOSNativeHandle.h
index df8efb9f..b074babe 100644
--- a/include/LLGL/Backend/OpenGL/MacOS/MacOSNativeHandle.h
+++ b/include/LLGL/Backend/OpenGL/MacOS/MacOSNativeHandle.h
@@ -8,8 +8,9 @@
 #ifndef LLGL_OPENGL_MACOS_NATIVE_HANDLE_H
 #define LLGL_OPENGL_MACOS_NATIVE_HANDLE_H
 
-
+#ifdef __OBJC__
 #include <Cocoa/Cocoa.h>
+#endif
 
 
 namespace LLGL
@@ -26,7 +27,11 @@ namespace OpenGL
 */
 struct RenderSystemNativeHandle
 {
+#ifdef __OBJC__
     NSOpenGLContext* context;
+#else
+    void* context;
+#endif
 };
 
 
diff --git a/include/LLGL/Backend/OpenGL/NativeHandle.h b/include/LLGL/Backend/OpenGL/NativeHandle.h
index 7a95392e..90aaddf1 100644
--- a/include/LLGL/Backend/OpenGL/NativeHandle.h
+++ b/include/LLGL/Backend/OpenGL/NativeHandle.h
@@ -21,9 +21,7 @@
 #   else
 #       include <OpenGL/gl.h>
 #   endif
-#   ifdef __OBJC__
-#       include <LLGL/Backend/OpenGL/MacOS/MacOSNativeHandle.h>
-#   endif
+#   include <LLGL/Backend/OpenGL/MacOS/MacOSNativeHandle.h>
 #elif defined(LLGL_OS_LINUX)
 #   include <GL/gl.h>
 #   include <LLGL/Backend/OpenGL/Linux/LinuxNativeHandle.h>
