diff --git a/CMakeLists.txt b/CMakeLists.txt
index 943f676b..623dc173 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,7 @@ endif()
 
 # === Build path ===
 
-set(LLGL_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/build" CACHE PATH "Override binary output directory")
+# set(LLGL_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/build" CACHE PATH "Override binary output directory")
 
 set( EXTERNAL_MODULE_DIR            "${PROJECT_SOURCE_DIR}/cmake"                   )
 set( PROJECT_INCLUDE_DIR            "${PROJECT_SOURCE_DIR}/include"                 )
diff --git a/include/LLGL/Platform/MacOS/MacOSNativeHandle.h b/include/LLGL/Platform/MacOS/MacOSNativeHandle.h
index ccbf599f..8db03b02 100644
--- a/include/LLGL/Platform/MacOS/MacOSNativeHandle.h
+++ b/include/LLGL/Platform/MacOS/MacOSNativeHandle.h
@@ -8,8 +8,11 @@
 #ifndef LLGL_MACOS_NATIVE_HANDLE_H
 #define LLGL_MACOS_NATIVE_HANDLE_H
 
-
+#ifdef __OBJC__
 #include <Cocoa/Cocoa.h>
+#else
+typedef void NSResponder;
+#endif
 
 
 namespace LLGL
