diff --git a/CMakeLists.txt b/CMakeLists.txt
index abb54505..fd08488d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -238,7 +238,7 @@ if(NOT COMMAND find_host_program)
 endif()
 
 # Tests require Python3
-find_host_package(Python3 REQUIRED)
+# find_host_package(Python3 REQUIRED)
 
 # Check for symbol exports on Linux.
 # At the moment, this check will fail on the OSX build machines for the Android NDK.

diff --git a/tools/diff/diff.cpp b/tools/diff/diff.cpp
index d60edb2e..284e22d4 100644
--- a/tools/diff/diff.cpp
+++ b/tools/diff/diff.cpp
@@ -87,8 +87,8 @@ std::unique_ptr<spvtools::opt::IRContext> load_module(const char* path) {
     return spvtools::BuildModule(
         kDefaultEnvironment, spvtools::utils::CLIMessageConsumer,
         std::string(contents.begin(), contents.end()),
-        spvtools::SpirvTools::kDefaultAssembleOption |
-            SPV_TEXT_TO_BINARY_OPTION_PRESERVE_NUMERIC_IDS);
+        (int)spvtools::SpirvTools::kDefaultAssembleOption |
+            (int)SPV_TEXT_TO_BINARY_OPTION_PRESERVE_NUMERIC_IDS);
   }
 
   std::vector<uint32_t> contents;
